Quotient/threadview.cpp

```diff
@@ -0,0 +1,77 @@
+// SPDX-FileCopyrightText: 2024 James Graham <james.h.graham@protonmail.com>
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#include "threadview.h"
+
+#include "events/roommessageevent.h"
+
+using namespace Quotient;
+
+void ThreadView::add(std::unique_ptr<Thread> thread)
+{
+    _threads.push_back(std::move(thread));
+}
+
+bool ThreadView::erase(const QString& threadRootId)
+{
+    auto threadIt = std::find_if(_threads.begin(), _threads.end(), [threadRootId](const auto& thread) { return thread->threadRootId() == threadRootId; });
+    if (threadIt == _threads.end()) {
+        return false;
+    }
+     _threads.erase(threadIt);
+    return true;
+}
+
+bool ThreadView::exisits(const QString& threadRootId) const
+{
+    return std::find_if(_threads.begin(), _threads.end(), [threadRootId](const auto& thread) { return thread->threadRootId() == threadRootId; }) != _threads.end();
```

Comment:
```suggestion
    return std::ranges::any_of(_threads, [threadRootId](const auto& thread) { return thread->threadRootId() == threadRootId; });
```
That `checkThreadRoot()` lambda factory might come handy here, too.